### DISCUSSION TITLE
Not having much luck emulating a Bash function.  Help?

### DISCUSSION THREAD

**Discussion by wolf** [xonsh/xonsh#4787] [Q&A] [answered]
```bash
function fcat() { # fcat <glob> : find all the files in or below . whose names match <glob> and cat them
    fd --follow --hidden --glob --type f "$@" --exec-batch bat
}
```

I would just use a Xonsh alias, but I need to stick arguments in the **middle** of the command I want to execute.  I've been reading everything I could get my hands on and have yet to figure this out.  It's got to be something simple I'm just missing!  What do I need to do?

**Comment by gforsyth** [comment, marked as answer]
If you want an alias that takes an argument you can create it like this:
```
def _fcat(args):
    fd --follow --hidden --glob --type f @(args[0]) --exec-batch bat

aliases["fcat"] = _fcat

# usage
fcat py  # --> fd --follow --hidden --glob --type f py --exec-batch bat
```

**Reply by wolf** [reply]
Could I get all the arguments with `@(*args)`?  And if so, would I have to do anything to the alias to allow multiple arguments?  And thanks so much for the response.  You are definitely setting me on the good path.

Example: `fcat --regex py`

Both the words following `fcat` would be inserted into the underlying call to `fd`.  That is
`fd --follow --hidden --glob --type f --regex py --exec-batch bat`

And also, loved your video.  I will be watching it again soon, when I know more :-)

**Reply by gforsyth** [reply]
You can use `@(args)` with no `*` and `xonsh` will unpack the arguments for you, e.g.

```
def _fcat(args):
    fd --follow --hidden --glob --type f @(args) --exec-batch bat

aliases["fcat"] = _fcat

fcat --regex py
```

No other changes needed for the alias to accept multiple arguments!

`args` is a list of arguments but you don't need to star-unpack them because that happens when we stringify to pass it to subprocess.

**Reply by wolf** [reply]
Thanks man.  I have a lot of these in my Bash environment and to give Xonsh a fair shake, I need to replicate a bunch them here.  This will really help!  And keep up the good work, by the way.  Xonsh is a fascinating shell.
